services/worker/src/worker/job_runners/dataset/is_valid.py

```diff
@@ -0,0 +1,73 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+from typing import Any, Literal, Mapping, Tuple, TypedDict
+
+from libcommon.constants import PROCESSING_STEP_DATASET_IS_VALID_VERSION
+from libcommon.simple_cache import SplitFullName, get_validity_by_kind
+
+from worker.job_runner import JobResult, JobRunner
+
+IsValidJobRunnerErrorCode = Literal[
+    "PreviousStepStatusError",
+    "PreviousStepFormatError",
+]
+
+
+class DatasetIsValidResponse(TypedDict):
+    valid: bool
+
+
+SPLIT_KINDS = [
+    "/splits",
+    "/split-names-from-streaming",
+    "/split-names-from-dataset-info",
+]
+FIRST_ROWS_KINDS = ["split-first-rows-from-streaming", "split-first-rows-from-parquet"]
+
+
+def compute_is_valid_response(dataset: str) -> Tuple[DatasetIsValidResponse, float]:
+    """
+    Get the response of /is-valid for one specific dataset on huggingface.co.
+
+    A dataset is valid if:
+    - /splits is valid
```

Comment:
One question:
```suggestion
    - /splits is valid for at least one of the splits 
```

Reply:
You're right. (nit: configs, not splits)

Reply:
done